src/components/ChatEntry.js

```diff
@@ -1,22 +1,46 @@
 import React from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp';
+import { useState } from 'react';
+
+
+const ChatEntry = ({sender, body, id, timeStamp, liked, updateChatData}) => {
+  const user ='Vladimir';
+  const [like, setLike] = useState(liked);
+  const userBubble = sender===user ? 'chat-entry local': 'chat-entry remote'
+
+  const handleLike = () => {
+      setLike(isLiked => {
+        const newState= !isLiked;
+        updateChatData(newState, id);
+        return newState;
+      })
+    };
+
+  const heart = like ? '❤️' : '🤍';
+  
+  
 
-const ChatEntry = (props) => {
   return (
-    <div className="chat-entry local">
-      <h2 className="entry-name">Replace with name of sender</h2>
+    <div className={`${userBubble}`}>
+      <h2 className="entry-name">{sender}</h2>
       <section className="entry-bubble">
-        <p>Replace with body of ChatEntry</p>
-        <p className="entry-time">Replace with TimeStamp component</p>
-        <button className="like">🤍</button>
+        <p>{body}</p>
+        <p className="entry-time"><TimeStamp time={timeStamp}/> </p>
+        <button onClick={handleLike} className="like">{heart}</button>
       </section>
     </div>
   );
 };
 
 ChatEntry.propTypes = {
-  //Fill with correct proptypes
+  id: PropTypes.number,
+  sender: PropTypes.string.isRequired,
+  body: PropTypes.string.isRequired,
+  timeStamp: PropTypes.string.isRequired,
+  liked: PropTypes.bool.isRequired,
+  updateChatData: PropTypes.func.isRequired
 };
```

Comment:
Nice prop types. Ideally, we would probably mark all of these props as required, but since there was some test weirdness with the props due to how the test data was provided, it's fine that they're not.

In general, if a property isn't mark required, we should also ensure that our component works properly in its absence. For example, if the callback isn't supplied, we might avoid calling it, or even avoid rendering the heart as clickable at all. Or if the liked value isn't supplied, we might render the message without the liked indicator at all.